server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -3925,8 +3926,12 @@ public void testScheduledRefresh() throws Exception {
         logger.info("--> ensure search idle");
         assertTrue(primary.isSearchIdle());
         assertTrue(primary.searchIdleTime() >= TimeValue.ZERO.millis());
+        long periodicFlushesBefore = primary.flushStats().getPeriodic();
         primary.flushOnIdle(0);
+        assertBusy(() -> assertThat(primary.flushStats().getPeriodic(), greaterThan(periodicFlushesBefore)));
+
         logger.info("--> scheduledRefresh(future5)");
+        ensureNoPendingScheduledRefresh();
```

Comment:
If there's some spurious refresh being run, we cannot be sure it is at this exact spot.

Maybe a better approach would be to get the number of external refreshes before the future5, and asserting that after the scheduled refresh, it's incremented by 1.

Reply:
@kingherc That's a great idea. I've replaced the hack with blocking the refresh thread pool with checking the refresh stats.

Reply:
I would like to understand why the test was failing, in the logs of the test failures you can see that the flush is executed _after_ the assertion trips, so I'm not convinced about the flush being the issue here